detectors/opentelemetry_resource_detector_aws/src/otel_resource_aws_ecs.erl

```diff
@@ -0,0 +1,140 @@
+-module(otel_resource_aws_ecs).
+
+-behavior(otel_resource_detector).
+
+-export([get_resource/1]).
+
+-include_lib("kernel/include/logger.hrl").
+-include_lib("opentelemetry_semantic_conventions/include/incubating/attributes/cloud_attributes.hrl").
+-include_lib("opentelemetry_semantic_conventions/include/incubating/attributes/container_attributes.hrl").
+-include_lib("opentelemetry_semantic_conventions/include/incubating/attributes/aws_attributes.hrl").
+
+-define(OS_ENV, "ECS_CONTAINER_METADATA_URI_V4").
+-define(DEFAULT_CGROUP_PATH, "/proc/self/cgroup").
+
+%% @private
+-spec get_resource(list()) -> otel_resource:t() | no_return().
+get_resource(Config) ->
+    otel_resource:create(fetch(Config)).
+
+fetch(Config) ->
+    case os:getenv(?OS_ENV) of
+        false ->
+          erlang:error('Not running in ECS environment, ECS_CONTAINER_METADATA_URI_V4 not set.');
+        _MetadataV4URL ->
+          [
+            {?CLOUD_PROVIDER, atom_to_list(?CLOUD_PROVIDER_VALUES_AWS)},
+            {?CLOUD_PLATFORM, atom_to_list(?CLOUD_PLATFORM_VALUES_AWS_ECS)} |
+            aws_ecs_attributes(Config)
+          ]
+    end.
+
+aws_ecs_attributes(Config) ->
+  {ok, Hostname} = inet:gethostname(),
+
+  ContainerIDAttributes =
+    case fetch_container_id_attributes(Config) of
+      {ok, ContainerIDAttributesResult} ->
+        ContainerIDAttributesResult;
+      {error, ContainerIDAttributesError} ->
+        ?LOG_ERROR("Failed to fetch Container ID attributes: ~s", [ContainerIDAttributesError]),
+        []
+    end,
+  
+  MetadataV4Attributes =
+    case fetch_metadata_v4_attributes() of
+      {ok, MetadataV4AttributesResult} ->
+        MetadataV4AttributesResult;
+      {error, MetadataV4AttributesError} ->
+        ?LOG_ERROR("Failed to fetch metadata attributes: ~s", [MetadataV4AttributesError]),
+        []
+    end,
+  
+  [{?CONTAINER_NAME, Hostname}] ++ ContainerIDAttributes ++ MetadataV4Attributes.
+
+fetch_container_id_attributes(Config) ->
+  Path = proplists:get_value(cgroup_path, Config, ?DEFAULT_CGROUP_PATH),
+  
+  case file:read_file(Path) of
+      {ok, Data} ->
+          Lines = binary:split(Data, <<"\n">>, [global]),
+          ContainerId = lists:foldl(fun
+            (Line, none) ->
+              leength,
```

Comment:
What is going on here?

Reply:
Good catch, not sure how that assigns got there.